pkgs/development/compilers/ghc/8.2.1.nix

```diff
@@ -1,29 +1,58 @@
-{ stdenv, lib, fetchurl, bootPkgs, perl, ncurses, libiconv, binutils, coreutils
+{ stdenv, fetchurl, bootPkgs, perl, binutils, coreutils
 , autoconf, automake, happy, alex, python3, sphinx, hscolour
-, buildPlatform, targetPlatform , selfPkgs, cross ? null
-
-  # If enabled GHC will be build with the GPL-free but slower integer-simple
+, ncurses, gmp, libffi
+, __targetPackages
+, buildPlatform, hostPlatform, targetPlatform
+
+, # LLVM is conceptually a run-time-only depedendency, but for
+  # non-x86, we need LLVM to bootstrap later stages, so it becomes a
+  # build-time dependency too.
+  #
+  # We generally want the latest llvm package set, which would normally be
+  # `llvmPackages` on most platforms. But on Darwin, the default is the version
+  # released with OSX, so we force 3.9, which is the correct version at this
```

Comment:
nixpkgs convention is to override versions like this at the callPackage site, not here.

Reply:
Now overridden in `pkgs/top-level/haskell-packags,nix`.